[Clive Ong] iP

cliveong opened the pull request:
## Duke bot
> Am terrible at coding, but am trying -Me 2021

This is a task manager with the following features
* Adding ToDo
* Adding Deadlines
* Adding Events

All you need to do is
1. Download the [Duke.jar file](https://github.com/cliveong/ip/releases)
2. Run it!

You can have **ALL OF IT** for ~a smol price of *$13.37*~ free! :grinning:

Stuff I still need to do here:
- [x] Finish week 3 requirements
- [x] Add gradle support
- [x] Add gui
- [ ] Actually make it nice

If you face any issues while using it `<☹ OOPS!!! I'm sorry>` will try to fix asap.

___
If you have read till here,

```java
public class Main {
    public static void main(String[] args) {
         System.out.println("Thank you");
    }
}
```

Commit:
Add Gradle support

cliveong committed:
Started level 1

cliveong committed:
Completed level 2

cliveong committed:
Level 2 bugfix

cliveong committed:
Level-3

cliveong committed:
Level-4

cliveong committed:
A-TextUiTesting

cliveong committed:
Added minimal exceptions, updated testing

cliveong committed:
Level-6

cliveong committed:
Bugfix

cliveong committed:
Complete level 7, use createDirectories and createNewFile in order to save the data in a folder called data

cliveong committed:
Revert "Complete level 7, use createDirectories and createNewFile in order to save the data in a folder called data"

This reverts commit 03da1b66908c3ff5578710e8ae2a75880253877e.

cliveong committed:
Revert "Revert "Complete level 7, use createDirectories and createNewFile in order to save the data in a folder called data""

This reverts commit 497864831dfce41b850f5ef925faecd9a64afc45.

cliveong committed:
Cut excess declaration

cliveong committed:
Finish Level-8

*Used localdate and accepts date in yyyy-mm-dd format

cliveong committed:
Merge branch 'branch-Level-7'

cliveong committed:
Merge branch 'branch-Level-8'

# Conflicts:
#	src/main/java/Duke.java

cliveong committed:
Add more oop

*Parser
*Storage
*TaskList
*Ui

cliveong committed:
Package all classes into duke

cliveong committed:
Add junit testing for exceptions and task

cliveong committed:
Remove excess variables, added MANIFEST.MF

cliveong committed:
Add Javadocs

cliveong committed:
Update coding style

cliveong committed:
Add find feature

cliveong committed:
Merge branch 'branch-A-JavaDoc'

cliveong committed:
Merge branch 'branch-A-CodingStandard'

# Conflicts:
#	src/main/java/duke/Event.java

cliveong committed:
Merge branch 'branch-Level-9'

# Conflicts:
#	src/main/java/duke/TaskList.java

trash-bin99 commented on src/main/java/duke/InvalidArguement.java:
```diff
@@ -0,0 +1,7 @@
+package duke;
+
+public class InvalidArguement extends DukeException{
```
Just a misspelling, InvalidArguement should be InvalidArgument.

ntwbruce replied:
Also needs a white space between DukeException and {.

cliveong replied:
Thanks, fixed

trash-bin99 commented on src/main/java/duke/Parser.java:
```diff
@@ -0,0 +1,67 @@
+        } catch (DukeException e) {
+            System.out.println(e.getMessage());
+        }
+    }
```
I think it would be useful to include some comments in between cases, to help guide the thought process for each condition.

cliveong replied:
Good idea

trash-bin99 commented on src/main/java/duke/Parser.java:
```diff
@@ -0,0 +1,67 @@
+     * @return validity
+     */
+    private boolean checkInvalidDescription(String input) {
+        if (input.trim().equals("todo") || input.trim().equals("deadline") || input.trim().equals("event")) {
```
This is perfectly valid, but I think it would be better to trim the input once beforehand then compare to the three cases:

```

String trimmedInput = input.trim();
if (trimmedInput.equals("todo") || trimmedInput.equals("deadline") || trimmedInput.equals("event")) {
    //rest of the code

```

trash-bin99 commented on src/main/java/duke/Storage.java:
```diff
@@ -0,0 +1,56 @@
+ * Manages the writing and creation of files duke.txt
+ */
+public class Storage {
+    String filepath = "src/main/data/duke.txt";
```
If this is a constant, might want to make it final and change the name to FILE_PATH.

trash-bin99 commented on src/main/java/duke/Task.java:
```diff
@@ -0,0 +1,32 @@
+        this.isDone = true;
+    }
+
+    abstract String logo();
```
Perhaps renaming it as getLogo() would sound better.

trash-bin99 reviewed (commented):
Hello!! Your code is relatively simple to understand and browse through, which is good! There are only a few issues I noticed, which I feel are relatively minor and easy to fix.

cliveong committed:
Merge remote-tracking branch 'origin/add-gradle-support'

ntwbruce commented on src/main/java/duke/DukeException.java:
```diff
@@ -0,0 +1,8 @@
+package duke;
+
+public class DukeException extends Exception{ //cs
```
I think you might need a white space separating Exception and {

ntwbruce commented on src/main/java/duke/Deadline.java:
```diff
@@ -0,0 +1,37 @@
+        return logo;
+    }
+
+    private String dateFormat() {
```
Methods should sound like verbs, so maybe something like formatDate().

ntwbruce commented on src/main/java/duke/Parser.java:
```diff
@@ -0,0 +1,67 @@
+                int number = Integer.parseInt(split[1]);
+                taskList.changeStatus(number);
+            } else if (input.length() > 9 && input.substring(0, 9).equals("deadline ")) { //date then time
+                String[] split = input.substring(9).split(" /by ");
```
Maybe not entirely necessary but it might be good to use more descriptive array names here, to complement comments like the one directly below explaining what the splitting does.

cliveong replied:
Good point!

ntwbruce commented on src/main/java/duke/Task.java:
```diff
@@ -0,0 +1,32 @@
+
+    public Task(String description) {
+        this.description = description;
+        this.isDone = false;
```
I think you don't need the 'this' keyword for isDone unless it's being shadowed by a constructor parameter (eg. if constructor had a param boolean isDone then you would do this.isDone = isDone).

ntwbruce reviewed (commented):
Looks good, just highlighted a few Java code standard issues. Also in general, JavaDoc descriptions should be full sentences, so I believe you need to add full stops. Remember to add descriptions for the parameters in the JavaDocs as well.

cliveong committed:
Complete level 10

cliveong committed:
Merge branch 'branch-Level-10'

LiuZhi1an commented on src/main/java/duke/Parser.java:
```diff
@@ -0,0 +1,86 @@
+        taskList.load();
+    }
+
+    /**
+     * Take in input from user and checks what commands to run
+     *
+     * @param input from user
+     */
+    public String commands(String input) {
+        try {
+            if (input.equals("list")) { //checks for list
+                return taskList.printList();
```
Perhaps can change the method name to something like readCommands or getCommands?

LiuZhi1an commented on src/main/java/duke/Storage.java:
```diff
@@ -0,0 +1,55 @@
+     *
+     * @param listOfText2
+     */
+    public void save(List<Task> listOfText2) {
+        String temp = "";
+        for (int i = 0; i < listOfText2.size(); i++) {
+            temp = temp + listOfText2.get(i).saveWithFormat() + "\n";
+        }
```
Changing to more descriptive names like saveToFile might be clearer.

cliveong replied:
Ty!

LiuZhi1an commented on src/main/java/duke/TaskList.java:
```diff
@@ -0,0 +1,146 @@
+    }
+
+    public void load() {
+        File existing = new File("src/main/data/duke.txt");
```
Maybe can change to names like currentFile instead of a boolean sounding name?

LiuZhi1an commented:
Good job! I think your code is pretty clear and well structured. I like how you credited the functions from the JavaFX tutorials.

cliveong committed:
Add assertion

Placed in the parser for command() to check if the commands that are
accepted is actually valid and not a bug in the string processing

*Checks for Index
*Checks for description

cliveong committed:
Merge pull request #1 from cliveong/branch-A-Assertions

Add assertion

cliveong committed:
Add Code Quality

*Cut down on long methods
*Reduce nesting > 4
*Make the happy path prominent
*Use default branch
*Minimise code duplication

cliveong committed:
Merge pull request #2 from cliveong/branch-A-CodeQuality

Add Code Quality

cliveong committed:
Add BCD-Extension

*C-DetectDuplicates

cliveong committed:
Merge branch 'branch-BCD-Extension'

cliveong committed:
Add gui changes

Add spacing and buffer area, overall still a wip

cliveong committed:
Add user guide via readme

cliveong committed:
Fix manifest and mainclassname

cliveong committed:
Rename Ui.png

cliveong committed:
Update README.md

cliveong committed:
Update README.md
